doc/Testing.md

```diff
@@ -0,0 +1,187 @@
+# Testing
+For more documentation on testing, read the references listed at the bottom.
+
+## Unit Testing
+- We use [flutter_test](https://api.flutter.dev/flutter/flutter_test/flutter_test-library.html) to create tests. You can create a test like this :
+    
+    ```bash
+    // Import the test package
+    import 'package:test/test.dart';
+    
+    void main() {
+      test('X should do something', () {
+        ...
+      });
+    }
+    ```
+    
+- To assert the result of a test, use `expect()`
+    
+    ```dart
+    /* 
+    void expect(
+      // actual value to be verified
+    	dynamic actual,
+      // characterises the expected result
+    	dynamic matcher, {
+    	// added in case of failure
+    	String? reason,
+    	// true or a String with the reasion to skip
+    	dynamic skip,
+    	}
+    )
+    */
+    
+    expect(actual, value);
+    expect(actual, isNull);
+    expect(actual, isNotNull);
+
+    ... and many more 
+    ```
+    
+    [Read more : Assertions in Dart and Flutter tests: an (almost) ultimate cheat sheet](https://invertase.io/blog/assertions-in-dart-and-flutter-tests-an-ultimate-cheat-sheet)
+    
+- We use [Mockito](https://pub.dev/packages/mockito) to mock classes.
+    
+    ```dart
+    import 'package:mocktail/mocktail.dart';
+    
+    // A Real Cat class
+    class Cat {
+      String sound() => 'meow!';
+      bool likes(String food, {bool isHungry = false}) => false;
+      final int lives = 9;
+    }
+    
+    // A Mock Cat class
+    class MockCat extends Mock implements Cat {}
+    
+    void main() {
+    	test('Cat should meow', () {
+    		// Arrange
+    	  final mock = MockCat();
+    		when(() => mock.sound()).thenReturn('meow!');
+    
+    		// Act
+    		mock.sound();
+    
+    		// Assert
+    		verify(() => mock.sound()).called(1);
+    		expect(mock.sound(), 'meow!');
+    	});	
+    }
+    ```
+    
+- We use [http-mock-adapter](https://pub.dev/packages/http_mock_adapter) to mock request-response communication with Dio.
+    
+    ```dart
+    void main() {
+    	test('get example', () async {
+    		// Arrange
+    		final dio = Dio(BaseOptions());
+    	  final dioAdapter = DioAdapter(dio: dio);
+    
+    	  const path = 'https://example.com';
+    
+    	  dioAdapter.onGet(
+    	    path,
+    	    (server) => server.reply(
+    	      200,
+    	      {'message': 'Success!'},
+    	      // Reply would wait for one-sec before returning data.
+    	      delay: const Duration(seconds: 1),
+    	    ),
+    	  );
+    
+    		// Act
+    	  final response = await dio.get(path);
+    
+    		// Assert
+    		expect(response.data, {'message': 'Success!'});
+    	});	
+    }
+    ```
+
+## Functional Testing
+As of now the template doesn't have functional tests. In a future update some examples will be provided.
```

Comment:
I think we should remove the part about functional testing if the template doesn't have functional tests yet. It's always best to document the current state of things and not the potential future.